localstack-core/localstack/services/stepfunctions/packages.py

```diff
@@ -18,6 +19,8 @@
 URL_ASPECTJWEAVER = f"{MAVEN_REPO_URL}/org/aspectj/aspectjweaver/1.9.7/aspectjweaver-1.9.7.jar"
 JAR_URLS = [URL_ASPECTJRT, URL_ASPECTJWEAVER]
 
+SFN_JAVA_VERSION = "11"
```

Comment:
@alexrashed By explicitly specifying the required JRE version in each downstream package, we remove the dependency on the value of fallback/default version in `java` package.

This will ensure that we can change the fallback/default version of `java` any time, without breaking any of the other packages.

Reply:
That's exactly the compromise. Either we can change the default version at any time but it's not really used anywhere (because it's explicitly set everywhere). Or we don't set the version wherever it's not explicitly necessary and we can change it in one place to test the upgrade of the default version everywhere with a minimal change.

This is why I would be in favor of only defining a Java version if we know it won't be compatible with Java 11+.

Reply:
I'm not sure I'm fully convinced, but I've removed the pins in c9faf3d0a99f6d9658f5ad21ae984ec50dfa4107. We can revisit this at a later point if need arises.

Reply:
I think we should come to an agreement though. Could you please elaborate why you are not convinced? I think we should discuss this and come to a unanimous conclusion for this kind of pattern as it's quite fundamental and would pave the way how we are going to handle this case in the future.

Reply:
That's an interesting analogy, but I would compare it more with our approach on any other dependency, i.e. our Python dependencies.

We don't really want to pin explicit dependencies, because it would make it harder for us to easily upgrade them. We only limit the range for dependencies where we know that there are incompatibilities. At the same time we do pin an exact version and once a week we run our action to update all lock files, creating a PR where we test the upgrade. If it works, it's great. If not, we can introduce certain version range limits or even exact pins.

For me the current approach here is the same. We don't pin the version in the specific installers (at least in those where we don't know about limitations). But we have the default version set (this would be the version lock in our analogy). We could then create a single PR to update the "lock" / default version and test if the integration works well. If it doesn't work, we can limit the version in certain installers or even move away from the upgrade completely.

In the future, I would also love to see an iteration on the package installer framework to (among other things) implement an automated process to suggest version bumps for package installers.